syft/pkg/cataloger/binary/default_classifiers.go

```diff
@@ -187,4 +187,24 @@ var defaultClassifiers = []classifier{
 		Package: "postgresql",
 		PURL:    mustPURL("pkg:generic/postgresql@version"),
 	},
+	{
+		Class:    "rust-standard-library-linux",
+		FileGlob: "**/libstd-*.so",
+		EvidenceMatcher: fileContentsVersionMatcher(
+			// clang LLVM (rustc version 1.48.0 (7eac88abb 2020-11-16))
+			`(?m)(\x00)clang LLVM \(rustc version (?P<version>[0-9]+(\.[0-9]+)?(\.[0-9]+)) \(\w+ \d{4}\-\d{2}\-\d{2}\)`),
+		Package: "rust",
+		PURL:    mustPURL("pkg:generic/rust@version"),
+		CPEs:    singleCPE("cpe:2.3:a:rust-lang:rust:*:*:*:*:*:*:*:*"),
+	},
+	{
+		Class:    "rust-standard-library-macos",
+		FileGlob: "**/libstd-*.dylib",
+		EvidenceMatcher: fileContentsVersionMatcher(
+			// c 1.48.0 (7eac88abb 2020-11-16)
+			`(?m)c (?P<version>[0-9]+(\.[0-9]+)?(\.[0-9]+)) \(\w+ \d{4}\-\d{2}\-\d{2}\)`),
```

Comment:
Is this the most specific we can be with the file globs? I wonder if `libstd-*.dylib` and `libstd-*.so` might match more than we really want? For the linux case, I think the `rustc version` is probably pretty specific, but this macOS pattern seems like it _might_ match things unexpectedly?

Reply:
No, that is the most specific possible

Reply:
The filenames are things like `libstd-d05561573e32b48f.dylib`

Reply:
The whole point being you can have many toolchain versions installed simultaneously

Reply:
I have a test image at https://github.com/anchore/test-images/blob/main/containers/analyzer-rust-toolchain/Dockerfile

Reply:
@kzantow , I tried `**/libstd-[0-9a-z][0-9a-z][0-9a-z][0-9a-z][0-9a-z][0-9a-z][0-9a-z][0-9a-z][0-9a-z][0-9a-z][0-9a-z][0-9a-z][0-9a-z][0-9a-z][0-9a-z][0-9a-z].dylib` but it doesn't seem to work

Reply:
It seems like that is valid for globstar but some logic in stereoscope search indexing is causing it to fail.  I'll look into it some more later, but perhaps @wagoodman might have ideas on it.

Reply:
that may be worth capturing as an issue in stereoscope, it should have at least fell through to the expensive glob case if it didn't know how to handle it (which is seems like it didn't?).